src/transform-sdk/go/transform/sr/serde.go

```diff
@@ -55,10 +61,84 @@ func DecodeFn[T any](fn func([]byte, T) error) SerdeOpt[T] {
 	return serdeOpt[T]{func(t *idSerde[T]) { t.decode = fn }}
 }
 
+type SubjectNameStrat int
+type TopicIndex int
+
+const (
+	TopicNameStrategy SubjectNameStrat = iota
+	RecordNameStrategy
+	TopicRecordNameStrategy
+)
+
+// ApplySubjectNameStrategy takes a topic name, a strategy, a suffix, and, optionally,
+// a fully qualified record name and uses those to derive a subject name.
+func ApplySubjectNameStrategy(topic string, strategy SubjectNameStrat, suffix string, name *string) (string, error) {
+	if strategy == TopicNameStrategy {
```

Comment:
I'm back and forth on this one.  I generally don't like using flags to control function behavior, but it also makes the behavior very clear for what's happening.   probably not an easier way to deal with this.  Also why is name a reference but the others aren't?

Reply:
fixed